plugin/octant.d.ts

```diff
@@ -113,31 +113,35 @@ export interface DashboardClient {
   /**
    * Get attempts to fetch a resource using the key provided
    * @param key - the key of the object to be fetched
+   * @param metadata - Optional arbitrary contexual data sent to octant
```

Comment:
These are kept optional to keep the existing plugins from becoming incompatible with octant.

Reply:
Do we have any guidance on what the content of the metadata is? An empty bag doesn't help a user to understand how to define the metadata. Perhaps `opts` is a better name

Reply:
Hey @scothis , metadata is essentially key/value store holding arbitrary data which is sent to octant. I don't have a strong preference for this name although opts will still be an empty bag. I called it metadata to match the name it is given when extracted from octant side. If you strongly prefer opts, please confirm and I will rename it as either of them works for me.

Reply:
I think both `opts` and `metadata` are not quite right. This data is only useful to a plugin author that is working with custom version of Octant that has implemented its own ObjectStore. For example a stepping Octant Playback object store for replaying a set of recorded Cluster and Octant events.

I think we want to clearly relay that for most people, you can and should ignore this. If I was using this library and say a field called `metadata` or `opts` I would want to go find out more about how I should use them.

Reply:
I'm leaning towards being verbose, explicit, and to the point of when I read this I immediately go "I don't care"

`@param customStoreContext - key/values to add to the context for object store requests, only useful for custom object stores`

Reply:
Another option here and it does duplicate some information, is to create a `CustomDashboardClient` or something like that .. that is documented but not the default so the TypeScript completion never even shows up for folks using the standard `DashboardClient`

Reply:
Just trying to get a sense of the cascading effects of defining a duplicate of `DashboardClient`. I'm guessing the plugin change inside `octant.d.ts` with default set to `DashboardClient` will be minimal but are we then conditionally picking one type(DashboardClient) or the other(CustomDashboardClient) when rendering the templates when using yeoman-generator. I guess a user will get an option to pick one or the other dashboard client when spinning up with `yeoman-generator`? Or every instance will be defined with something with type `DashboardClient | CustomDashboardClient`. I like the idea of having it separately defined to not affect the community members but It feels like regardless of which way this is implemented, there will be a point when we will need to document and clarify the use of the `customStoreContext` so it feels like just adding an extra parameter should be good enough?